packages/itwin/tree-widget/src/components/trees/models-tree/internal/ModelsTreeVisibilityHandler.ts

```diff
@@ -698,29 +693,68 @@ class ModelsTreeVisibilityHandlerImpl implements HierarchyVisibilityHandler {
    * Updates visibility of an element and all its child elements by adding them to the always/never drawn list.
    * @note If element is to be enabled and model is hidden, it will be enabled.
    */
-  private changeElementState(props: ChangeGeometricElementDisplayStateProps): Observable<void> {
+  private changeElementsState(props: ChangeGeometricElementsDisplayStateProps): Observable<void> {
     const result = defer(() => {
-      const { elementId, on, modelId, categoryId } = props;
+      const { elementIds, on, modelId, categoryId } = props;
       const viewport = this._props.viewport;
       return concat(
         props.on && !viewport.view.viewsModel(modelId) ? this.showModelWithoutAnyCategoriesOrElements(modelId) : EMPTY,
         defer(() => {
-          const categoryVisibility = this.getDefaultCategoryVisibilityStatus({ categoryId, modelId });
+          const categoryVisibility = this.getDefaultCategoryVisibilityStatus({ categoryId, modelId, ignoreTooltip: true });
           const isDisplayedByDefault = categoryVisibility.state === "visible";
-          return of(elementId).pipe(this.changeElementStateNoChildrenOperator({ on, isDisplayedByDefault }));
+          return this.queueElementChange(elementIds, on, isDisplayedByDefault);
         }),
       );
     });
-    return createVisibilityHandlerResult(this, props, result, this._props.overrides?.changeElementState);
+    return createVisibilityHandlerResult(this, props, result, this._props.overrides?.changeElementsState);
+  }
+
+  private queueElementChange(elementIds: Id64Set, on: boolean, visibleByDefault: boolean) {
```

Comment:
I'd rename this to `queueElementsVisibilityChange`

Reply:
Renamed